File: flutter_sdk/lib/src/ui/chat/view_models/image/image_state.dart
// Copyright (c) Yalochat, Inc. All rights reserved.

import 'package:chat_flutter_sdk/src/domain/models/image/image_data.dart';
import 'package:equatable/equatable.dart';

enum ImageStatus { initial, errorPickingImage, errorCancellingPick }

final class ImageState extends Equatable {
  final ImageData? pickedImage;
  final ImageData? hiddenImagePick;
  final ImageStatus imageStatus;

  const ImageState({this.pickedImage, this.hiddenImagePick, this.imageStatus = ImageStatus.initial});

  ImageState copyWith({
    ImageData? Function()? pickedImage,
    ImageData? Function()? hiddenImagePick,
    ImageStatus? imageStatus,
  }) {
    return ImageState(
      pickedImage: pickedImage != null ? pickedImage() : this.pickedImage,
      hiddenImagePick: hiddenImagePick != null ? hiddenImagePick() : this.hiddenImagePick,
      imageStatus: imageStatus ?? this.imageStatus,
    );
  }

  @override
  List<Object?> get props => [pickedImage, imageStatus, hiddenImagePick];
}

File: android-sdk/sdk/src/main/kotlin/com/yalo/chat/sdk/ui/chat/ImageState.kt
// Copyright (c) Yalochat, Inc. All rights reserved.

package com.yalo.chat.sdk.ui.chat

import com.yalo.chat.sdk.domain.model.ImageData

// Port of flutter-sdk/lib/src/ui/chat/view_models/image/image_state.dart
internal data class ImageState(
    // Image selected from gallery or captured by camera, awaiting user confirmation.
    val pickedImage: ImageData? = null,
    // True while the preview overlay is shown above the chat input.
    val isPreviewVisible: Boolean = false,
)
